ui/admin/app/components/worker-filter-generator/index.hbs

```diff
@@ -0,0 +1,198 @@
+        @value={{or (get @model @name) ''}}
+        @options={{hash
+          mode='text/x-sh'
+          gutters=(array 'CodeMirror-lint-markers')
+          lint='true'
+        }}
+      />
+    </Rose::CodeEditor>
+  </F.Control>
+  {{#if (get @model.errors @name)}}
+    <F.Error data-test-worker-filter-error as |E|>
+      {{#each (get @model.errors @name) as |error|}}
+        <E.Message>{{error.message}}</E.Message>
+      {{/each}}
+    </F.Error>
+  {{/if}}
+</Hds::Form::Field>
+
+<Hds::Separator />
+
+<Hds::Form::Toggle::Field
+  name='show_filter_generator'
+  {{on 'change' this.toggleFilterGenerator}}
+  as |F|
+>
+  <F.Label>{{t 'worker-filter-generator.toggle.title'}}</F.Label>
+  <F.HelperText>{{t
+      'worker-filter-generator.toggle.description'
+    }}</F.HelperText>
+</Hds::Form::Toggle::Field>
+
+<Hds::Separator />
+
+{{#if this.showFilterGenerator}}
+  <Hds::Form::Radio::Group
+    class='filter-generator-selection'
+    @name='filter_generator'
+    {{on 'change' this.setGeneratorType}}
+    as |G|
+  >
+    <G.Legend>{{t 'worker-filter-generator.filter-generator.title'}}</G.Legend>
+    <G.HelperText>
+      {{t 'worker-filter-generator.filter-generator.description'}}
+      <br />
+      <Hds::Link::Inline
+        @href={{doc-url 'worker-filters-format'}}
+        @color='secondary'
+      >
+        {{t 'worker-filter-generator.filter-generator.link'}}
+      </Hds::Link::Inline>
+    </G.HelperText>
+    <G.RadioField
+      @value={{this.generatorTagType}}
+      checked={{eq this.selectedGeneratorType this.generatorTagType}}
+      as |F|
+    >
+      <F.Label>{{t
+          'worker-filter-generator.filter-generator.tag.label'
+        }}</F.Label>
+      <F.HelperText>{{t
+          'worker-filter-generator.filter-generator.tag.helper'
+        }}</F.HelperText>
+    </G.RadioField>
+    <G.RadioField
+      @value={{this.generatorNameType}}
+      checked={{eq this.selectedGeneratorType this.generatorNameType}}
+      as |F|
+    >
+      <F.Label>{{t
+          'worker-filter-generator.filter-generator.name.label'
+        }}</F.Label>
+      <F.HelperText>{{t
+          'worker-filter-generator.filter-generator.name.helper'
+        }}</F.HelperText>
+    </G.RadioField>
+  </Hds::Form::Radio::Group>
+
+  {{#let (unique-id) (unique-id) as |labelId helpId|}}
+    <Hds::Form::Fieldset
+      aria-labelledby={{labelId}}
+      aria-describedby={{helpId}}
+      class='input-values'
+      as |F|
+    >
+      <F.Legend id={{labelId}}>{{t
+          'worker-filter-generator.filter-generator.input-values.title'
+        }}</F.Legend>
+      <F.HelperText id={{helpId}}>{{t
+          'worker-filter-generator.filter-generator.input-values.description'
+        }}</F.HelperText>
+      <F.Control>
+        {{#if (eq this.selectedGeneratorType this.generatorTagType)}}
+          <Hds::Form::TextInput::Field
+            @value={{this.key}}
+            name='tag_key'
+            @width='320px'
```

Comment:
None blocking comments: I see the selects all defining a `@width='320px'` totally fine with it, but wondering, since all of them are within `<F.Control>` adding a width to the container will bring same effect.

Reply:
I played around with the suggestion but it seems `<F.Control>` does not like getting a `@width` attribute.